### PR TITLE
vcd/resource_vcd_vapp_network.go: Check for RESOLVED status for powering off vApp

### LINKED ISSUE #1091: vcd_vapp_network/vcd_vapp_org_network: reboot_vapp_on_removal does not work correctly on vApp status "Resolved"
### Terraform Version
`Terraform v1.4.4`

### Affected Resource(s)
Please list the resources as a list, for example:
- vcd_vapp_network
- vcd_vapp_org_network

### Terraform Configuration Files

```hcl
resource "vcd_vapp" "vapp" {
  name = "vapptest1"

  lease {
    runtime_lease_in_sec = 0
    storage_lease_in_sec = 0
  }
}

resource "vcd_vapp_org_network" "vappnetattach" {
  vapp_name = "vapptest1"

  org_network_name = "network-1"
  reboot_vapp_on_removal = true
}
```

### Debug Output
```
2023/07/20 13:35:43 [TRACE] reboot_vapp_on_removal=true is enabled with parent vApp 'vapptest1
[...]
2023/07/20 13:35:44 [TRACE] reboot_vapp_on_removal=true, vApp 'vapptest1' status before network removal is 'RESOLVED'
2023/07/20 13:35:44 [TRACE] reboot_vapp_on_removal=true, powering off vApp
```

### Expected Behavior
When destroying above configuration this should work without any errors.

### Actual Behavior
When destroying above configuration the provider tries to power off the vApp which does not work because it is in status"RESOLVED" and ends up in following error:
> Error: error Powering Off: error undeploy vApp: API Error: 400: [ xxxxx ] The requested operation could not be executed since vApp "vapptest1" is not running.

### Steps to Reproduce
1. `terraform apply`
2. `terraform destroy`

### Important Factoids
vApps are in status "RESOLVED" if there are no VMs assigned to it. So the problem occurs if you want to delete an "empty" vApp or even if the VM got deleted before by Terraform itself.

### PR DESCRIPTION
**Pull request by NilsBusche**
Fixes problem described in #1091 
Closes #1091

### CODE REVIEW

**Comment by Didainius** [comment]
@NilsBusche ,
Thanks for the PR!. I'd like to clarify when do you get it to `RESOLVED` state?
I will have to reinforce it with a test so want to find out how to replicate the problem

**Comment by NilsBusche** [comment]
@Didainius vApp gets into RESOLVED, if you create it without any VMs in it or if you have a stopped vApp and remove all VMs from it.

**Comment by Didainius** [comment]
Thanks! 

I have created a test that fails without your patch. The stupid part is that I can't make a PR to your repository as in the end it will probably ruin CLAbot (it doesn't like PRs that have multiple contributors). 
Therefore I have put the patch to a gist - https://gist.github.com/Didainius/59b072055eb30949eed84dfe6c6a8621

Could you do a `git apply patch.git` after downloading that file. After it is here, we should be able to run a broader test suite to verify that nothing got broken along the way.

**Comment by NilsBusche** [comment]
I've added the patch

**Review by Didainius — APPROVED** [review]
Thanks. I am good with it. I ran tests with `vm vapp network` tags on 10.3.3 (which wouldn't be impacted anyway) and 10.4.2.

**Review by dataclouder — CHANGES_REQUESTED** [review]

File: vcd/resource_vcd_vapp_network_test.go [hunk 1]
```diff
@@ -1205,3 +1205,85 @@ func TestAccVcdNsxtVappNetworkRemovalFails(t *testing.T) {
...
+func TestAccVcdNsxtVappNetworkRemovalFailsResolvedVapp(t *testing.T) {
+	preTestChecks(t)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	vcdClient := createTemporaryVCDConnection(false)
+	if vcdClient.Client.APIVCDMaxVersionIs("< 37.1") {
```
[the remark this answers is not in the document]
Comment (NilsBusche) [reply vcd/resource_vcd_vapp_network_test.go]: Changed

File: vcd/resource_vcd_vapp_network_test.go [hunk 2]
```diff
@@ -1205,3 +1205,85 @@ func TestAccVcdNsxtVappNetworkRemovalFails(t *testing.T) {
...
 	postTestChecks(t)
 }
+
+func TestAccVcdNsxtVappNetworkRemovalFailsResolvedVapp(t *testing.T) {
+	preTestChecks(t)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
```
[the remark this answers is not in the document]
Comment (NilsBusche) [reply vcd/resource_vcd_vapp_network_test.go]: Changed

File: vcd/resource_vcd_vapp_network_test.go [hunk 3]
```diff
@@ -1205,3 +1205,85 @@ func TestAccVcdNsxtVappNetworkRemovalFails(t *testing.T) {
...
+		"ExistingRoutedNetwork": testConfig.Nsxt.RoutedNetwork,
+		"RebootVappOnRemoval":   "true",
+
+		"Tags": "network vapp",
+	}
+	testParamsNotEmpty(t, params)
+
+	configText1 := templateFill(testAccVcdNsxtVappNetworkRemovalResolvedVapp, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 1: %s", configText1)
+
```
[the remark this answers is not in the document]
Comment (NilsBusche) [reply vcd/resource_vcd_vapp_network_test.go:1230]: Changed

**Review by dataclouder — APPROVED** [review]
Thanks for your contribution!

**Review by lvirbalas — APPROVED** [review]
Thank you for contributing!

**Review by adezxc — APPROVED** [review]

**Comment by Didainius** [comment]
Thank you @NilsBusche , this is merged and set for `3.11`

**Pull request merged**